### ISSUE TITLE
negative death

### ISSUE THREAD

**Issue by SourJay** [suriyun-production/mmorpg-kit-docs#797]
https://discord.com/channels/460801727789137920/460803281455939595/810908955421310976
this issue still exists
https://gyazo.com/b120494e26df49b6a7e07f4547d31c9a

**Comment by insthync** [comment]
Any error logs?

**Comment by SourJay** [comment]
NullReferenceException: Object reference not set to an instance of an object
MultiplayerARPG.AnimatorCharacterModel.PlayMoveAnimation () (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Model/3D/AnimatorCharacterModel.cs:592)
MultiplayerARPG.BaseCharacterModel.SetMovementState (MultiplayerARPG.MovementState movementState, MultiplayerARPG.ExtraMovementState extraMovementState, UnityEngine.Vector2 direction2D) (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Model/BaseCharacterModel.cs:611)
MultiplayerARPG.BaseGameEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BaseGameEntity.cs:455)
MultiplayerARPG.VehicleEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/VehicleSystems/VehicleEntity.cs:133)
MultiplayerARPG.BaseGameEntity.Update () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BaseGameEntity.cs:421)

**Comment by SourJay** [comment]
NullReferenceException: Object reference not set to an instance of an object
MultiplayerARPG.DamageableEntity.ReceivedDamage (UnityEngine.Vector3 fromPosition, MultiplayerARPG.EntityInfo instigator, MultiplayerARPG.CombatAmountType combatAmountType, System.Int32 damage, MultiplayerARPG.CharacterItem weapon, MultiplayerARPG.BaseSkill skill, System.Int16 skillLevel) (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/DamageableEntity.cs:176)
MultiplayerARPG.BaseCharacterEntity.ReceivedDamage (UnityEngine.Vector3 fromPosition, MultiplayerARPG.EntityInfo instigator, MultiplayerARPG.CombatAmountType combatAmountType, System.Int32 damage, MultiplayerARPG.CharacterItem weapon, MultiplayerARPG.BaseSkill skill, System.Int16 skillLevel) (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterEntity/BaseCharacterEntity.cs:499)
MultiplayerARPG.DefaultGameplayRule.ApplyFallDamage (MultiplayerARPG.BaseCharacterEntity character, UnityEngine.Vector3 lastGroundedPosition) (at Assets/UnityMultiplayerARPG/Core/Scripts/GameData/Rule/DefaultGameplayRule.cs:713)
MultiplayerARPG.BaseCharacterEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterEntity/BaseCharacterEntity.cs:250)
MultiplayerARPG.BasePlayerCharacterEntity.EntityUpdate () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/CharacterEntity/PlayerCharacterEntity/BasePlayerCharacterEntity.cs:72)
MultiplayerARPG.BaseGameEntity.Update () (at Assets/UnityMultiplayerARPG/Core/Scripts/Gameplay/BaseGameEntity.cs:421)

**Comment by SourJay** [comment]
i made a flying vehicle that just basically jumps and glides but if i exit mid air and take fall damage it goes negative and does not give me a option to respawn unless i hit esc

**Comment by SourJay** [comment]
https://gyazo.com/1a37c8a0667f5fb6cd0c0b142383548a character is in idle positon taking this damage https://gyazo.com/917d3ab35d14c6e2b85db6b8377e00e7

**Comment by insthync** [comment]
Try changes codes in `DefaultGameplayRule.cs` -> function: `ApplyFallDamage` 
Line 713 to 
```
character.ReceivedDamage(character.CacheTransform.position, new EntityInfo(), CombatAmountType.NormalDamage, damage, null, null, 0);
```

![Screenshot_208](https://user-images.githubusercontent.com/2697550/108170078-3664af80-712c-11eb-851f-9c3c78a8ee29.png)

**Comment by insthync** [comment]
Tell me if code changes above are work or not, It was included in 1.62e
